### PR TITLE
Parse polygon kml.

### PR DESCRIPTION
**Pull request by EBartusiak**
- Parse polygon ring and polygon.
- Refactor verifyPolygonRing() to be in ParsingHelpers.

### CODE REVIEW

File: Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/KMLParser.java [hunk 1]
```diff
@@ -195,10 +199,79 @@ private MapPolyline parseLineString()
...
+    boolean hasOuterBoundary = false;
+    AltitudeReferenceSystemWrapper altitudeReferenceSystemWrapper =
+        new AltitudeReferenceSystemWrapper(AltitudeReferenceSystem.GEOID);
+    while (mParser.next() != XmlPullParser.END_TAG) {
+      if (mParser.getEventType() != XmlPullParser.START_TAG) {
+        continue;
+      }
+      String type = mParser.getName();
+      if (type.equals("outerBoundaryIs") || type.equals("innerBoundaryIs")) {
+        if (type.equals("outerBoundaryIs")) {
```
Comment (KristofferHenriksson) [inline Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/KMLParser.java:220]: Is it ok to have 2 inner boundaries? If so a comment describing why outer boundary is checked for to only occur once is helpful.

File: Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/KMLParser.java [hunk 2]
```diff
@@ -195,10 +199,79 @@ private MapPolyline parseLineString()
...
+          verifyElementNotSeen("outerBoundaryIs", hasOuterBoundary);
+          hasOuterBoundary = true;
+        }
+        ArrayList<Geoposition> positions = parsePolygonRing(type, altitudeReferenceSystemWrapper);
```
Comment (KristofferHenriksson) [inline Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/KMLParser.java:224]: No need for named local variable, can just pass parsePolygonRing result to rings.add()

File: Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/KMLParser.java [hunk 3]
```diff
@@ -195,10 +199,79 @@ private MapPolyline parseLineString()
...
+    boolean hasParsedCoordinates = false;
+    while (mParser.next() != XmlPullParser.END_TAG) {
+      if (mParser.getEventType() != XmlPullParser.START_TAG) {
+        continue;
+      }
+      String type = mParser.getName();
+      if (type.equals("coordinates")) {
+        verifyElementNotSeen(type, hasParsedCoordinates);
+        positions = parseCoordinates(altitudeReferenceSystemWrapper);
+        String exceptionMessage = ParsingHelpers.verifyPolygonRing(positions);
```
Comment (KristofferHenriksson) [inline Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/KMLParser.java:260]: The naming of verifyPolygonRing is confusing. All the other verify methods throw an exception if it's not valid and this one doesn't; it instead returns a string that if not null is the error message if the input is invalid.

File: Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/KMLParser.java (hunk 1 above)
Comment (EBartusiak) [reply Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/KMLParser.java:220]: Yes; one outer boundary is required, and 0 or more inner boundaries are allowed. I'll add a comment.

File: Android/Samples/libraries/moduletools/src/main/java/com/microsoft/maps/moduletools/ParsingHelpers.java [hunk 4]
```diff
@@ -26,4 +27,54 @@ public static void logAltitudeWarning() {
...
+            .append(", ");
+      }
+      positionsStringBuilder.append(positions.get(positions.size() - 1));
+      return "Polygon ring must have at least 4 positions, and the first and last position must "
+          + "be the same. Instead saw Geopositions: ["
+          + positionsStringBuilder
+          + "].";
+    }
+
+    Geoposition firstPosition = positions.get(0);
```
Comment (MaciejMS) [inline Android/Samples/libraries/moduletools/src/main/java/com/microsoft/maps/moduletools/ParsingHelpers.java:55]: This is again not clear for me why are you certain that this element exists. Please add check or a comment.

Comment (EBartusiak) [reply Android/Samples/libraries/moduletools/src/main/java/com/microsoft/maps/moduletools/ParsingHelpers.java:55]: For this method, there is a check added that there are at least 4 elements (enough for a valid polygon) in the list positions.

Comment (EBartusiak) [reply Android/Samples/libraries/moduletools/src/main/java/com/microsoft/maps/moduletools/ParsingHelpers.java:55]: Check added that there is at least one element in the list (for printing the Geopositions).

**Pull request merged**